_pytest/fixtures.py

```diff
@@ -1068,6 +1068,9 @@ def parsefactories(self, node_or_obj, nodeid=NOTSET, unittest=False):
         self._holderobjseen.add(holderobj)
         autousenames = []
         for name in dir(holderobj):
+            # Skip properties before the following getattr executes them.
+            if isinstance(getattr(type(holderobj), name, None), property):
```

Comment:
I will switch from `getattr(type(holderobj), name, None)` to `type(holderobj).__dict__.get(name)`, which is hopefully safer. Here I assume that `type(X)` has a `__dict__` for all `X`. I think this is true - even if I try something like this:

```
>>> class FooMeta(type):
...    __slots__ = ()
... 
>>> class Foo(metaclass=FooMeta):
...   pass
```

it still has a `__dict__`:

```
>>> type(Foo())
<class '__main__.Foo'>
>>> type(Foo()).__dict__
mappingproxy({...stuff...})
```